### PR TITLE
Allow numbers as filenames while adding datasets to library folder

### PR DESCRIPTION
**Pull request by OlegZharkov**
If the filename of dataset is a number, it will raise an exception. This PR allows filenames to be numbers

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/library_datasets.py [hunk 1]
```diff
@@ -402,6 +402,8 @@ def load(self, trans, payload=None, **kwd):
         if path is None:
             raise exceptions.RequestParameterMissingException('The required attribute path is missing.')
         folder = self.folder_manager.get(trans, folder_id)
+        if isinstance(path, int):
+            path = str(path)
```
Comment (mvdbeek) [inline lib/galaxy/webapps/galaxy/api/library_datasets.py]: I don't know, I think this should be a string when passed in from the client, and I wonder if this doesn't hide something else going wrong there.

File: lib/galaxy/webapps/galaxy/api/library_datasets.py [hunk 2]
```diff
@@ -401,8 +401,8 @@ def load(self, trans, payload=None, **kwd):
         path = kwd.get('path', None)
         if path is None:
             raise exceptions.RequestParameterMissingException('The required attribute path is missing.')
-        if isinstance(path, int):
-            raise exceptions.RequestParameterInvalidException('The required attribute path is Integer.')
+        if not isinstance(path, str):
+            raise exceptions.RequestParameterInvalidException('The required attribute path is not String.')
```
Comment (mvdbeek) [inline lib/galaxy/webapps/galaxy/api/library_datasets.py:400]:
```suggestion
            raise exceptions.RequestParameterInvalidException('The required attribute path is not a string.')
```

**Comment by OlegZharkov** [comment]
@mvdbeek 
Does it look better now? It will be always String now, Python won't cast it as Integer.
I just added a check on backend: "raise an exception if path is not a String"

**Review by mvdbeek — APPROVED** [review]

**Comment by mvdbeek** [comment]
Yep, thanks for digging!

**Comment by OlegZharkov** [comment]
> Yep, thanks for digging!

a sure thing, thanks a lot for reviewing it!

File: lib/galaxy/webapps/galaxy/api/library_datasets.py (hunk 2 above)
Comment (dannon) [reply lib/galaxy/webapps/galaxy/api/library_datasets.py:400]: @OlegZharkov Looks like the article got dropped after the rebase?

**Comment by bgruening** [comment]
Thanks @OlegZharkov!

**Pull request merged**